File: fivpt.visual/jr2as.f
      function jr2as(re,lbuf,icn,it,id,isbuf)
      integer*2 lbuf(1)
C
      jr2as=ir2as(re,lbuf,icn,it,id)
      if (.not.jchar(lbuf,icn).eq.36) return
      ita=min0(iabs(it)+iabs(id)+1,isbuf*2-icn+1)
      jr2as=ir2as(re,lbuf,icn,ita,id)
      jr2as=iabs(it)

      return
      end

File: onoff/jr2as.c
#include <stdlib.h>
#include <string.h>

void jr2as(re,lbuf,it,id,isbuf)
     float re;
     char lbuf[];
     int it,id,isbuf;
{
  int ita, icn;

  icn=strlen(lbuf);
  flt2str(lbuf,re,it,id);
  if(lbuf[icn]!='$')
    return;
  ita=abs(it)+abs(id)+1;
  ita=ita<isbuf-(icn+1)?ita:isbuf-(icn+1);
  lbuf[icn]=0;
  flt2str(lbuf,re,ita,id);
  lbuf[icn+abs(it)]=0;
}     
